swarm/api/http/server_test.go

```diff
@@ -892,3 +894,239 @@ func TestMethodsNotAllowed(t *testing.T) {
 	}
 
 }
+
+// HTTP convenience function
+func httpDo(httpMethod string, url string, reqBody io.Reader, headers map[string]string, verbose bool, t *testing.T) (*http.Response, string) {
+	// Build the Request
+	req, _ := http.NewRequest(httpMethod, url, reqBody)
+	for key, value := range headers {
+		req.Header.Set(key, value)
+	}
+	if verbose {
+		t.Log(req.Method, req.URL, req.Header, req.Body)
+	}
+
+	// Send Request out
+	httpClient := &http.Client{}
+	res, err := httpClient.Do(req)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Read the HTTP Body
+	buffer, _ := ioutil.ReadAll(res.Body)
+	defer res.Body.Close()
+	body := string(buffer)
+
+	return res, body
+}
+
+func TestGet(t *testing.T) {
+	verbose := false
+	// Setup Swarm and upload a test file to it
+	srv := testutil.NewTestSwarmServer(t, serverFunc)
+	defer srv.Close()
+
+	// Accept: text/html GET / -> 200 HTML, Swarm Landing Page
+	getRootHTML := func() {
```

Comment:
regarding the separation of tests through these closures:
it might be cleaner to create an anonymous struct, create a slice which contains all of the parameters, methods and assertions and just iterate over them and execute one at a time.

for example:
```
	getRootHTML := func() { } ... ()
	getRootJSON := func() { } ... ()
```
would be converted into:
``` go
for _, v := range []struct{
   verbose bool
   method string
   uri string
   headers map[string]string
   expectedStatusCode int
   matchReponseBody string
   expectedStatusCode int
} {
   { 
      verbose: false,
      method: "GET",
      headers: map[string]string{
           "Accept": "text/html"
      }
     .....
   },
} { 
		res, body := httpDo("GET", fmt.Sprintf("%s/%s", srv.URL,v.uri), nil, v.headers, v.verbose, t)
 		if res.StatusCode != v.expectedStatusCode {
			t.Fatal("got response code %d, expected %d", res.StatusCode, v.expectedStatusCode)
		}
		if !strings.HasPrefix(body, v.matchReponseBody) {
			t.Fatal("expected response body to containt: %s", v.matchReponseBody)
		}
}
```

I would, however, split the multipart test into a different function since the test structure is a bit different and won't really conform with my suggestion above (or would require a more generalised implementation).
you could also of course pull out the function that iterates over the struct elements one level higher, at your own accord.